trie-db/src/lib.rs

```diff
@@ -377,52 +403,69 @@ where
 /// Trait with definition of trie layout.
 /// Contains all associated trait needed for
 /// a trie definition or implementation.
-pub trait TrieLayout {
+/// Structure implementing this trait can define contextual behavior
+/// for trie, but should be small (clone cost should be insignifiant).
+pub trait TrieLayout: Default + Clone {
 	/// If true, the trie will use extension nodes and
 	/// no partial in branch, if false the trie will only
 	/// use branch and node with partials in both.
 	const USE_EXTENSION: bool;
 	/// If true, the trie will allow empty values into `TrieDBMut`
 	const ALLOW_EMPTY: bool = false;
+
 	/// Hasher to use for this trie.
 	type Hash: Hasher;
 	/// Codec to use (needs to match hasher and nibble ops).
 	type Codec: NodeCodec<HashOut=<Self::Hash as Hasher>::Out>;
+
+	/// Threshold above which an external node should be
+	/// use to store a node value.
+	fn max_inline_value(&self) -> Option<u32>;
```

Comment:
out of curiosity why not using a const here ?

Reply:
It is mostly related to past design.
With current substrate usage, I can effectively choose a different layout when calling storage root, state-machine should allow that.
Then I just have to use two different layouts.
This would mean less change in substrate side (no need to pass around an instantiated layout boolean).
I think I will start a branch with the change and update if ok, with current code it will not fondamentally change things, but making a less permissive api may be good.